### PR TITLE
[Merged by Bors] - increase the maximum number of point lights with shadows to the max supported by the device

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Being limited to 10 pointlights with shadow is very limiting

## Solution

- Raise the limit

### CODE REVIEW

File: crates/bevy_pbr/src/render/light.rs [hunk 1]
```diff
@@ -600,6 +599,12 @@ pub fn prepare_lights(
 
     let mut point_lights: Vec<_> = point_lights.iter().collect::<Vec<_>>();
 
+    let nb_shadow_lights = point_lights
+        .iter()
+        .filter(|light| light.1.shadows_enabled)
+        .count()
+        .min(MAX_POINT_LIGHT_SHADOW_MAPS);
```
Comment (superdump) [inline crates/bevy_pbr/src/render/light.rs:606]: We have to also respect the max texture array layers limit here:
```suggestion
    let nb_shadow_lights = point_lights
        .iter()
        .filter(|light| light.1.shadows_enabled)
        .count()
        .min(MAX_POINT_LIGHT_SHADOW_MAPS)
        .min(render_device.limits().max_texture_array_layers / 6);
```

**Review by superdump — CHANGES_REQUESTED** [review]

File: crates/bevy_pbr/src/render/light.rs [hunk 2]
```diff
@@ -600,6 +593,15 @@ pub fn prepare_lights(
 
     let mut point_lights: Vec<_> = point_lights.iter().collect::<Vec<_>>();
 
+    #[cfg(not(feature = "webgl"))]
+    let max_point_light_shadow_maps = point_lights
+        .iter()
+        .filter(|light| light.1.shadows_enabled)
+        .count()
+        .min((render_device.limits().max_texture_array_layers / 6) as usize);
+    #[cfg(feature = "webgl")]
```
Comment (superdump) [inline crates/bevy_pbr/src/render/light.rs:602]: I thought this kind of `#[cfg()]` on an expression was a nightly feature?

File: crates/bevy_pbr/src/render/light.rs [hunk 3]
```diff
@@ -600,6 +593,15 @@ pub fn prepare_lights(
...
+    let max_point_light_shadow_maps = 1;
+
     // Sort point lights with shadows enabled first, then by a stable key so that the index can be used
     // to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows.
```
Comment (superdump) [inline crates/bevy_pbr/src/render/light.rs:606]: You missed this reference in a comment. There are some others too.

File: crates/bevy_pbr/src/render/light.rs (hunk 2 above)
Comment (mockersf) [reply crates/bevy_pbr/src/render/light.rs:602]: well... no

**Review by superdump — APPROVED** [review]

File: crates/bevy_pbr/src/render/light.rs [hunk 4]
```diff
@@ -600,8 +593,17 @@ pub fn prepare_lights(
 
     let mut point_lights: Vec<_> = point_lights.iter().collect::<Vec<_>>();
 
+    #[cfg(not(feature = "webgl"))]
+    let max_point_light_shadow_maps = point_lights
+        .iter()
+        .filter(|light| light.1.shadows_enabled)
+        .count()
+        .min((render_device.limits().max_texture_array_layers / 6) as usize);
```
Comment (IceSentry) [inline crates/bevy_pbr/src/render/light.rs:601]: What's the 6 here and on line 624? I'm not sure what it's supposed to mean

Comment (superdump) [reply crates/bevy_pbr/src/render/light.rs:601]: True. It is a ‘magic number’ I guess though it’s already been used elsewhere so I’d fix it in a separate PR. Point lights use cubemaps for shadow mapping. Cubemaps are array textures with 6 layers per cubemap, one for each cube face. That’s where the 6 comes from.

**Review by IceSentry — APPROVED** [review]

**Review by cart — APPROVED** [review]

**Pull request closed without merging**